components/subcomponents/movieNight/eventReminder.py

```diff
@@ -0,0 +1,62 @@
+import asyncio
+from datetime import datetime, timedelta, timezone
+import pickle
+import discord
+from common.asyncTask import AsyncTask
+import common.utils as ut
+
+REMINDER_THRESHOLD_SECONDS = 1 * 60 * 60
+eventReminderTask: AsyncTask | None = AsyncTask(lambda: __remind_event_coroutine())
+LAST_EVENT_PATH = "./database/last_event_start_time.pkl"
+
+
+# Sends reminder to everyone with MOVIE_ROLE REMINDER_THRESHOLD_SECONDS before the event starts
+# Only set up reminders if:
+#   1) event exists
+#   2) event is scheduled
+#   3) event occurs in the future
+#   4) a reminder was not already sent for this event
+def start_event_reminder():
+    event: discord.ScheduledEvent | None = ut.get_movie_event()
+    if not event:
+        return
+
+    is_event_sent: bool
+    try:
+        with open(LAST_EVENT_PATH, "rb") as file:
+            last_event_start_time = pickle.load(file)
+            is_event_sent = isinstance(last_event_start_time, datetime) and last_event_start_time == event.start_time
+    except FileNotFoundError:
+        is_event_sent = False
+    if event.status is discord.EventStatus.scheduled and datetime.now(timezone.utc) <= event.start_time and not is_event_sent:
+        eventReminderTask.start()
+
+
+# Coroutine to remind everyone with MOVIE_ROLE REMINDER_THRESHOLD_SECONDS before the event starts
+# Runs in the background task in the following situations:
+#   1) every time the ScheduledEvent MOVIE_EVENT_NAME changes
+#   2) every time the application restarts
+async def __remind_event_coroutine():
+    try:
+        # Set reminder for REMINDER_THRESHOLD_SECONDS before event starts
+        event: discord.ScheduledEvent = ut.get_movie_event()
+        reminder_threshold = event.start_time - timedelta(seconds=REMINDER_THRESHOLD_SECONDS)
+        delta = reminder_threshold - datetime.now(timezone.utc)
+        await asyncio.sleep(delta.total_seconds())
+
+        # Ensure that currently it is REMINDER_THRESHOLD_SECONDS before the event is supposed to start.
+        reminder_threshold = event.start_time - timedelta(seconds=REMINDER_THRESHOLD_SECONDS)
+        now = datetime.now(timezone.utc)
+        if not reminder_threshold <= now <= event.start_time:
+            return
```

Comment:
Do we lose a reminder if this is not true? If so, good to log this error in console or Discord.

This makes me think that we should have a logging class for devs. Will add to backlog

Reply:
The code is essentially used to stop repeated reminders. If the bot randomly restarts after sending the initial reminder (an hour before movie-night), it will send another reminder even though (I think) it should not.

Added logging as requested.